src/test/java/ezschedule/logic/parser/ShowNextCommandParserTest.java

```diff
@@ -0,0 +1,66 @@
+package ezschedule.logic.parser;
+
+import static ezschedule.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static ezschedule.logic.commands.ShowNextCommand.SHOW_UPCOMING_COUNT_ONE;
+import static ezschedule.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static ezschedule.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import org.junit.jupiter.api.Test;
+
+import ezschedule.logic.commands.ShowNextCommand;
+import ezschedule.model.event.UpcomingEventPredicate;
+
+/**
+ * As we are only doing white-box testing, our test cases do not cover path variations
+ * outside of the ShowNextCommand code. For example, inputs "1" and "1 abc" take the
+ * same path through the ShowNextCommand, and therefore we test only one of them.
+ * The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
+public class ShowNextCommandParserTest {
+
+    private final ShowNextCommandParser parser = new ShowNextCommandParser();
+
+    @Test
+    public void parse_noArgs_returnsShowNextCommandOneUpcoming() {
+        final ShowNextCommand expected = new ShowNextCommand(new UpcomingEventPredicate(SHOW_UPCOMING_COUNT_ONE));
+
+        assertParseSuccess(parser, "", expected);
+        assertParseSuccess(parser, "   ", expected);
+    }
+
+    @Test
+    public void parse_validArgs_returnsShowNextCommandArbitraryUpcoming() {
+        assertParseSuccess(parser, "5", new ShowNextCommand(new UpcomingEventPredicate(5)));
+        assertParseSuccess(parser, " 7 ", new ShowNextCommand(new UpcomingEventPredicate(7)));
+        assertParseSuccess(parser, " 33", new ShowNextCommand(new UpcomingEventPredicate(33)));
+        assertParseSuccess(parser, "500", new ShowNextCommand(new UpcomingEventPredicate(500)));
+    }
```

Comment:
Would it be possible for there to be comments on the equivalence partitions? Similar to `parse_invalidArgs_throwsParseException()`.

Reply:
The equivalence partition can be inferred from the test name i.e. arbitrary numbers which are valid arguments.
To better understand the difference between the different asserts in here, I have added some comments of their differences.